app/models/event.rb

```diff
@@ -0,0 +1,16 @@
+class Event < ApplicationRecord
+  belongs_to :eventable, polymorphic: true
+
+  belongs_to :creator, class_name: "User", optional: true
+  belongs_to :target, class_name: "User", optional: true
+
+  validates :action, presence: true
+
+  def description
+    if creator && !target
+      "#{action.humanize(capitalize: false)} by #{creator.name}"
```

Comment:
kinda hard to mentally parse with no context - maybe I should add an example or two as a comment for why it's formatted this way?

```ruby
# eg. approved by Matt about an hour ago / Matt added Gary less than a minute ago
```

Reply:
There might also be edge cases in the future where certain types of events should be phrased differently. This will (hopefully) cover the uses and we can always implement a system for optional overrides later.

Reply:
yep that makes sense! are we adding the time distance elsewhere? or should it be included in this method?

Reply:
In `timelines/show.html.erb` -> `_event.html.erb` (or however we end up doing it), I was thinking something like:

```erb
<%= event.description %> <%= local_time_ago(event.created_at) %>
```

with the [local_time](https://github.com/basecamp/local_time) gem.